[devel] save_download_pstate AssertionError

<!-- Delete non-relevant parts if this is not a bug report -->

#### Please, fill all relevant items: ####

- [X] I have read CONTRIBUTING.md

- [X] I have tried with the latest pre-release version and I still can reproduce the issue.

##### Tribler version/branch+revision: #####
3aa6bb204bbcc0cf7be3d6e8a355baf544ef6cb6

##### Behavior: #####
When you close Tribler when it is still initializing downloads it will throw an uncaught AssertionError.

```
ERROR   1476869025.44  LaunchManyCore:703   Exception while checkpointing: MyTestTorrent
Traceback (most recent call last):
  File "/home/quinten/tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py", line 700, in network_checkpoint_callback
    self.save_download_pstate(infohash, pstate)
  File "/home/quinten/tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py", line 867, in save_download_pstate
    self.downloads[infohash].save_resume_data())
  File "/home/quinten/tribler/tribler/Tribler/dispersy/taskmanager.py", line 38, in register_task
    assert isinstance(task, (Deferred, DelayedCall, LoopingCall)), (task, type(task) == type(Deferred))
AssertionError: (None, False)

```
##### Cause: #####
`LibtorrentDownloadImpl.save_resume_data()` is annotated as `@waitForHandleAndSynchronize()`: this means the default return value is `None` if it cannot synchronize/is shutting down.

##### Fix: #####
Simply check if `self.downloads[infohash].save_resume_data()` is not `None` before registering the task in `LaunchManyCore.save_download_pstate`.

--- a/Tribler/Core/APIImplementation/LaunchManyCore.py
+++ b/Tribler/Core/APIImplementation/LaunchManyCore.py
@@ -784,9 +784,11 @@ class TriblerLaunchMany(TaskManager):
         """ Called by network thread """
 
         self.downloads[infohash].pstate_for_restart = pstate
+        save_resume_data = self.downloads[infohash].save_resume_data()
 
-        self.register_task("save_pstate %f" % timemod.clock(),
-                           self.downloads[infohash].save_resume_data())
+        if save_resume_data:
+            self.register_task("save_pstate %f" % timemod.clock(),
+                               save_resume_data)
 
     def load_download_pstate(self, filename):
         """ Called by any thread """

